src/helper-modules/ordnance-survey/ordnance-survey.service.test.ts

```diff
@@ -0,0 +1,125 @@
+import { HttpService } from '@nestjs/axios';
+import { ConfigService } from '@nestjs/config';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { AxiosError } from 'axios';
+import { when } from 'jest-when';
+import { of, throwError } from 'rxjs';
+import expectedResponseData = require('./examples/example-response-for-search-places-v1-postcode.json');
+import noResultsResponseData = require('./examples/example-response-for-search-places-v1-postcode-no-results.json');
+
+import { GEOSPATIAL } from '@ukef/constants';
+
+import { OrdnanceSurveyException } from './exception/ordnance-survey.exception';
+import { OrdnanceSurveyService } from './ordnance-survey.service';
+
+describe('OrdnanceSurveyService', () => {
+  const valueGenerator = new RandomValueGenerator();
+
+  let httpServiceGet: jest.Mock;
+  let configServiceGet: jest.Mock;
+  let service: OrdnanceSurveyService;
+
+  const testPostcode = GEOSPATIAL.EXAMPLES.POSTCODE;
+  const testKey = valueGenerator.string({ length: 10 });
+  const basePath = '/search/places/v1/postcode';
+
+  const expectedResponse = of({
+    data: expectedResponseData,
+    status: 200,
+    statusText: 'OK',
+    config: undefined,
+    headers: undefined,
+  });
+
+  beforeEach(() => {
+    const httpService = new HttpService();
+    const configService = new ConfigService();
+    httpServiceGet = jest.fn();
+    httpService.get = httpServiceGet;
+
+    configServiceGet = jest.fn().mockReturnValue({ key: testKey });
+    configService.get = configServiceGet;
+
+    service = new OrdnanceSurveyService(httpService, configService);
+  });
+
+  describe('getAddressesByPostcode', () => {
+    const expectedPath = `${basePath}?postcode=${encodeURIComponent(testPostcode)}&lr=EN&key=${encodeURIComponent(testKey)}`;
+
+    const expectedHttpServiceGetArgs: [string, object] = [expectedPath, { headers: { 'Content-Type': 'application/json' } }];
+
+    describe.each([
+      {
+        postcode: testPostcode,
+        expectedUrlQueryPart: `?postcode=${encodeURIComponent(testPostcode)}`,
+      },
+      {
+        postcode: 'W1A 1AA',
+        expectedUrlQueryPart: `?postcode=W1A%201AA`,
+      },
+      {
+        postcode: 'W1A1AA',
+        expectedUrlQueryPart: '?postcode=W1A1AA',
```

Comment:
Can we please add following assertions:
1. Scottish post code search
2. Welsh post code search
3. NI post code search
4. Invalid post code searches:
  * ''
  * []
  * 'ABC'

Reply:
User input is validated in router, before controller. In this service we just to URL encoding using encodeURIComponent, so '', 'ABC' and other strings would be valid input. 

123, false, [] is stopped by typescript type check.